### DISCUSSION TITLE
Creating a static Zephyr repo with all submodules for pipelines creates a problem with west extensions.

### DISCUSSION THREAD

**Discussion by Erdyke** [zephyrproject-rtos/zephyr#52041] [Q&A]
For automating builds that run in a docker image I'd like to create a fix Zephyr repository that does not need to be initialized every time a new pipeline runs. I've removed the .git folders for each submodule which allows me to commit all the required files in a static way.

Having removed the .git folder removes west extension commands. Specifically I'm interested in the build command. If I restore the .git folder, the extension commands return. Is there a way around requiring the .git folder?

For reference I'm working out of the nRF connect 1.9.1 SDK that utilizes Zephyr.

**Comment by yashi** [comment]
The `build` extension is under `zephyr/scripts/west_commands`.  How does removing `.git/` folder from _sub-modules_ remove it?

BTW, I'm just curious, how do you _update_ your static Zephyr repo?  Do you commit the files from sub-module to the Zephyr repo? If so, why not use [T2 topology](https://docs.zephyrproject.org/3.1.0/develop/west/workspaces.html#t2-star-topology-application-is-the-manifest-repository)?

With T2 topology combined with `--narrow` or `-o=--depth=1` to `west update`, you
- don't need a special setup
- don't have to pull all sub-modules Zephyr proper needs ([a sample](https://github.com/yashi/micro-ros-zephyr-demo/blob/main/west.yml#L21) and [the doc](https://docs.zephyrproject.org/3.1.0/develop/west/manifest.html#option-3-mapping))
- don't have to pull all history of the repos. ([discussion](https://github.com/zephyrproject-rtos/west/issues/319))

**Comment by Erdyke** [comment]
Hi Yashi, thanks for the reply.

Your question about the `.git` folder is part of my question too. I don't understand why it matters, unless `west` cares that it is in a git repository folder.

Probably part of my problem is that the repository I'm working with does not have a recommend zephyr workspace topology. Instead we're bootstrapping out environment by exporting ZEPHYR_BASE to the environment and then running west commands.

I was hoping for a way to resolve this problem without restructuring the project tree.

**Reply by yashi** [reply]
I can't reproduce. I've created a new zephyr workspace from scratch and removed `.git/` from one of the sub-modules, picolibc, but `west build` is still there.

```
$ rm -rf modules/lib/picolibc/.git 
$ rm -rf build && west build -b qemu_cortex_m3 zephyr/samples/hello_world/
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/yashi/foo/zephyr/samples/hello_world
-- Found Python3: /usr/bin/python3.10 (found suitable exact version "3.10.8") found components: Interpreter 
-- Cache files will be written to: /home/yashi/.cache/zephyr
-- Zephyr version: 3.2.99 (/home/yashi/foo/zephyr)
-- Found west (found suitable version "0.14.0", minimum required is "0.7.1")
-- Board: qemu_cortex_m3
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.15.1 (/home/yashi/zephyr-sdk-0.15.1)
-- Found toolchain: zephyr 0.15.1 (/home/yashi/zephyr-sdk-0.15.1)
-- Found Dtc: /usr/bin/dtc (found suitable version "1.6.1", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/yashi/foo/zephyr/boards/arm/qemu_cortex_m3/qemu_cortex_m3.dts
-- Generated zephyr.dts: /home/yashi/foo/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/yashi/foo/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /home/yashi/foo/build/zephyr/dts.cmake
Parsing /home/yashi/foo/zephyr/Kconfig
Loaded configuration '/home/yashi/foo/zephyr/boards/arm/qemu_cortex_m3/qemu_cortex_m3_defconfig'
Merged configuration '/home/yashi/foo/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/home/yashi/foo/build/zephyr/.config'
Kconfig header saved to '/home/yashi/foo/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/yashi/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /home/yashi/foo/build
-- west build: building application
[1/135] Preparing syscall dependency handling

[2/135] Generating include/generated/version.h
-- Zephyr version: 3.2.99 (/home/yashi/foo/zephyr), build: zephyr-v3.2.0-1517-g21c4bbe385c1
[125/135] Linking C executable zephyr/zephyr_pre0.elf

[129/135] Linking C executable zephyr/zephyr_pre1.elf

[135/135] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:        8594 B       256 KB      3.28%
             RAM:        3936 B        64 KB      6.01%
```

You said  removing `.git/` from sub-modules removed `west build`.  But you are also using `ZEPHYR_BASE` with non-recommended topology? Would you please give us a minimal working example?

**Comment by Erdyke** [comment]
Ah, I see. It looks like the problem specifically arises when I remove the git folder from zephyr's directory.
